go/array_string/unique/is_unique.go

```diff
@@ -0,0 +1,20 @@
+// Package unique contains the implementation of
+// an algorithm that  checks the characters of a
+// string are unique with time complexity of O(n)
+// and space complexity amortized O(1) - O(n)
+// if the whole string is an ASCII, we'll be dealing
+// with 128 characters and if it is Unicode, it will be more.
+package unique
+
+func isUnique(words string) bool {
+	visited := make(map[rune]bool)
+
+	for _, char := range words {
+		// if we have seen the character before
```

Comment:
I like to think of comments as explaining "why" but never "what". If you have to explain what you are doing then the code probably needs to be fixed up (e.g. rename variables or even totally refactored). If you have a reason for making a decision then it usually can't be communicated in the code itself so should be a comment.

TL;DR this comment isn't necessary